### PR TITLE
adding cert option to some checks and metrics

### PR DESCRIPTION
**Pull request by csoleimani**
## Pull Request Checklist

**Is this in reference to an existing issue?**
kind of. https://github.com/sensu-plugins/sensu-plugins-elasticsearch/issues/100 

it's better to pass in a cert than not check for validation

#### General

- [ ] Update Changelog following the conventions laid out [here](https://github.com/sensu-plugins/community/blob/master/HOW_WE_CHANGELOG.md)

- [ ] Update README with any necessary configuration snippets

- [ ] Binstubs are created if needed

- [ ] RuboCop passes

- [ ] Existing tests pass

#### New Plugins

- [ ] Tests

- [ ] Add the plugin to the README

- [ ] Does it have a complete header as outlined [here](http://sensu-plugins.io/docs/developer_guidelines.html#coding-style)

### CODE REVIEW

**Comment by majormoses** [comment]
Let me know if you need help working through the rubocop errors.

**Comment by csoleimani** [comment]
Sure, some help would be great. I tried running the autofix flag, but it just created some more issues. I'm not very familiar with Ruby and don't know all the best practices

**Comment by majormoses** [comment]
I went threw the cops one by one, in some cases they were rather opinionated other places we should certainly refactor at a later date.

File: Rakefile [hunk 1]
```diff
@@ -36,4 +36,4 @@ task :check_binstubs do
   end
 end
 
-task default: [:spec, :make_bin_executable, :yard, :rubocop, :check_binstubs]
+task default: %i(spec make_bin_executable yard rubocop check_binstubs)
```
Comment (majormoses) [inline Rakefile]: pretty sure this changes in a newer version of rubocop, but that change is I believe breaking so we can change that back when we update it.

File: bin/check-es-indices-sizes.rb [hunk 2]
```diff
@@ -159,12 +159,11 @@ def build_indices_with_sizes
   def run
     node_fs_stats = client.nodes.stats metric: 'fs,indices'
     nodes_being_used = node_fs_stats['nodes'].values.select { |node| node['indices']['store']['size_in_bytes'] > 0 }
-    # rubocop:disable SingleLineBlockParams
-    # rubocop:disable LineLength
-    used_in_bytes = nodes_being_used.map { |node| node['fs']['data'].map { |data| data['total_in_bytes'] - data['available_in_bytes'] }.flatten }.flatten.inject { |sum, x| sum + x }
-    # rubocop:enable LineLength
+    # TODO: come back and cleanup all these rubocop disables with a little refactor
```
Comment (majormoses) [inline bin/check-es-indices-sizes.rb]: I know this is not the code that was changed but somehow rubocop started complaining. This is super ugly and hard to read, we should come back and refactor this later.

**Comment by majormoses** [comment]
Thanks for your contribution to Sensu plugins! Without people like you submitting PRs we couldn't run the project. I will review it shortly.

**Comment by majormoses** [comment]
@csoleimani any chance on coming back to get this cleaned up and over the line?

**Comment by majormoses** [comment]
@thomasriley thanks for testing I will review this tonight or tomorrow and hopefully merge and release this.

**Comment by majormoses** [comment]
Looks like we need a rebase.

**Comment by majormoses** [comment]
I rebased for you, sorry for the long delay.

**Comment by majormoses** [comment]
To avoid confusion I am gonna rename the option from `--cert` to `--cert-file` to make it clear that this supports passing a file name not the actual cert directly via cli.

**Comment by thomasriley** [comment]
@majormoses I agree on both using the `--cert-file` name to ensure that option is descriptive and avoid using the short `-c` to avoid confusion and maintain consistency.

**Comment by majormoses** [comment]
released: https://rubygems.org/gems/sensu-plugins-elasticsearch/versions/1.10.0

**Comment by jwsomis** [comment]
Why was this not added to all checks?

**Comment by majormoses** [comment]
@jwsomis cant say but you are welcome to open up a PR for any missing ones.

**Comment by csoleimani** [comment]
I think I had some problems with the ones I didn't add it to, but it was so long ago that I don't remember what they are. Feel free to give it a shot, though!

**Comment by jwsomis** [comment]
I started trying to, but am rather new to sensu plugins and ran into issues
trying to figure out what the correct build steps are for bonsai assets.
Could you enlighten me?

Wes

**Comment by csoleimani** [comment]
This is the best guide I have seen around sharing assets:

https://discourse.sensu.io/t/contributing-assets-for-existing-ruby-sensu-plugins/1165

**Comment by jwsomis** [comment]
Thanks, Cameron! I was missing the TravisCI portion.

**Pull request merged**